tensorflow_quantum/python/layers/circuit_executors/unitary_test.py

```diff
@@ -0,0 +1,105 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for tensorflow_quantum.layers.circuit_executors.unitary."""
+import numpy as np
+from absl.testing import parameterized
+import sympy
+import tensorflow as tf
+import cirq
+
+from tensorflow_quantum.python.layers.circuit_executors import unitary
+from tensorflow_quantum.python import util
+
+
+class UnitaryTest(parameterized.TestCase, tf.test.TestCase):
+    """Basic tests for the State layer."""
+
+    def test_unitary_create(self):
+        """Test that State layers can be created."""
+        _ = unitary.Unitary()
+
+    def test_basic_inputs(self):
+        """Test that State layer outputs work end to end."""
+        simple_circuit = cirq.Circuit(cirq.H(cirq.GridQubit(0, 0)))
+        cirq_u = cirq.unitary(simple_circuit)
+        tfq_u = unitary.Unitary()(simple_circuit)
+        self.assertAllClose(tfq_u, [cirq_u])
+
+    def test_single_circuit_batch_inputs(self):
+        """Test that a single circuit with multiple parameters works."""
+        a_symbol = sympy.Symbol('alpha')
+        some_values = np.array([[0.5], [3.5]])
+        circuit = cirq.Circuit(cirq.H(cirq.GridQubit(0, 0))**a_symbol)
+        results = unitary.Unitary()(circuit,
+                                    symbol_names=[a_symbol],
+                                    symbol_values=some_values)
+        u_1 = cirq.unitary(
+            cirq.resolve_parameters(circuit, {a_symbol: some_values[0][0]}))
+        u_2 = cirq.unitary(
+            cirq.resolve_parameters(circuit, {a_symbol: some_values[1][0]}))
+        print(u_1)
+        self.assertAllClose(results, [u_1, u_2])
+
+    def test_multi_circuit_batch(self):
+        """Test that a batch of circuits works."""
+        a_symbol = sympy.Symbol('alpha')
+        some_values = np.array([[0.5], [3.5]])
+        circuit = cirq.Circuit(cirq.H(cirq.GridQubit(0, 0))**a_symbol)
+        results = unitary.Unitary()(util.convert_to_tensor([circuit, circuit]),
+                                    symbol_names=[a_symbol],
+                                    symbol_values=some_values)
+        u_1 = cirq.unitary(
+            cirq.resolve_parameters(circuit, {a_symbol: some_values[0][0]}))
+        u_2 = cirq.unitary(
+            cirq.resolve_parameters(circuit, {a_symbol: some_values[1][0]}))
+        print(u_1)
+        self.assertAllClose(results, [u_1, u_2])
+
+    def test_input_errors(self):
```

Comment:
With the other layers, tests of `input_check.py` were moved to `input_checks_test.py` to avoid test duplication; should we do that here, and leave just one test to be sure `expand_circuits` gets called?

Reply:
I separated the op errors and keras errors. The tests are pretty small so I'm fine with leaving them as is for now.